lib/Logger/LogAppenderFile.cpp

```diff
@@ -269,19 +272,20 @@ void LogAppenderFile::reopenAll() {
 void LogAppenderFile::closeAll() {
   for (auto& it : _fds) {
     int fd = std::get<0>(it);
-    // set the fd to "disabled"
-    std::get<0>(it) = -1;
-    // and also tell the appender of the file descriptor change
-    std::get<2>(it)->updateFd(-1);
-
     if (fd > STDERR_FILENO) {
+      // set the fd to "disabled"
+      std::get<0>(it) = -1;
+      // and also tell the appender of the file descriptor change
+      std::get<2>(it)->updateFd(-1);
```

Comment:
I don't know if this change is ok...
Let's assume the server is a daemon and closed fds 0 and 1 on start.
Then fds 0 and 1 could be reused for any other file, even for custom logfiles. I think we need to do the `updateFd` call unconditionally of the file descriptor number.
I fear the file handling in the LogAppender* needs some more cleanup.

Reply:
I will try to figure out what it currently does and then try to untangle the LogAppender* objects from the somehow managing the global list of appenders.